pages/report.js

```diff
@@ -0,0 +1,253 @@
+    });
+  } else {
+    // Remove all fields except the `@id` column
+    query.deleteKeyValue("field");
+    query.addKeyValue("field", "@id");
+  }
+  return query.format();
+};
+
+const Report = ({ searchResults }) => {
+  const router = useRouter();
+  const { profiles } = useContext(SessionContext);
+
+  // Ref of the scrollable table DOM <div> so we can detect scroll position
+  const gridRef = useRef(null);
+
+  const { path, queryString } = splitPathAndQueryString(searchResults["@id"]);
+  const sortedColumnId = getSortColumn(searchResults);
+  const defaultColumnSpecs = schemaColumnsToColumnSpecs(searchResults.columns);
+  const reportType = getReportType(searchResults);
+  const allColumnSpecs = profiles
+    ? getReportTypeColumnSpecs(reportType, profiles)
+    : [];
+  const nonSortableColumnIds = searchResults.non_sortable || [];
+
+  /**
+   * Navigate to the same page but with the "sort=" query parameter set to the column that was
+   * clicked. This also handles reverse sorting when the user clicks on the currently sorted
+   * column.
+   * @param {string} columnId Property name of the column that was clicked
+   */
+  const onHeaderCellClick = (columnId) => {
+    const query = new QueryString(queryString);
+    const newSortColumn =
+      columnId === sortedColumnId ? `-${columnId}` : columnId;
+    query.replaceKeyValue("sort", newSortColumn);
+    router.push(`${path}?${query.format()}`);
+  };
+
+  /**
+   * Called when the user clicks an individual column's visibility toggle in the modal. Depending
+   * on the current state of the "field=" query parameter, this function will either add or remove
+   * the column's property name from the query string.
+   * @param {string} columnId ID of the column the user made visible or invisible in the modal
+   * @param {boolean} isVisible True if the column is now visible, false if it is now invisible
+   */
+  const onColumnVisibilityChange = (columnId, isVisible) => {
+    const updatedQueryString = updateColumnVisibilityQuery(
+      queryString,
+      columnId,
+      isVisible,
+      defaultColumnSpecs
+    );
+    router.push(`${path}?${updatedQueryString}`);
+  };
+
+  /**
+   * Called when the user chooses to hide or show all columns in the modal at once. When hiding all
+   * columns, the `@id` column remains visible.
+   * @param {boolean} isAllVisible True to make all possible columns visible, false to hide all
+   */
+  const onAllColumnsVisibilityChange = (isAllVisible) => {
+    const updatedQueryString = updateAllColumnsVisibilityQuery(
+      queryString,
+      isAllVisible,
+      allColumnSpecs
+    );
+    router.push(`${path}?${updatedQueryString}`);
+  };
+
+  if (profiles) {
+    const pageTitle = composeSearchResultsPageTitle(searchResults, profiles);
+    const items = searchResults["@graph"];
+    const columns = generateColumns(searchResults, profiles);
+
+    return (
+      <>
+        {pageTitle && <PagePreamble pageTitle={pageTitle} />}
```

Comment:
should display Breadcrumbs on the top to make it consistent.

Reply:
Yeah, that’s true. I’ve added it to a list for a future ticket.

Reply:
Sounds good